src/AutoRest.CSharp/Generation/Writers/MgmtResponseHeaderGroupWriter.cs

```diff
@@ -10,12 +10,12 @@
 
 namespace AutoRest.CSharp.Generation.Writers
 {
-    internal class ResponseHeaderGroupWriter
+    internal class MgmtResponseHeaderGroupWriter
```

Comment:
There is a lot of duplication between this and management writer, can we share code?

Reply:
I was following your pattern, sharing code in most of these places requires more templatization which didn't seem like a route that was desirable.  If we do this we can do it in a later PR

Reply:
There are other ways to do it than templetization. Base types for example.

Reply:
Or helper methods

Reply:
Agreed, but for now I am just following the pattern that is there

Reply:
What do you mean? I don't think we have entire copies of code files anywhere in feature/v3.

Reply:
I'd point to RestClientBuilder as an example of using a bit of templates and a builder to not copy paste all of that code between writers.